pkg/db/migrations/202311151850_add_resources.go

```diff
@@ -0,0 +1,28 @@
+package migrations
+
+import (
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+)
+
+func addResources() *gormigrate.Migration {
+	type Resource struct {
+		Model
+		ConsumerID string         `gorm:"index"`
+		Version    int            `gorm:"not null"`
+		Spec       datatypes.JSON `gorm:"type:json"`
+		Status     datatypes.JSON `gorm:"type:json"`
+	}
+
+	return &gormigrate.Migration{
+		ID: "202311151850",
+		Migrate: func(tx *gorm.DB) error {
+			return tx.AutoMigrate(&Resource{})
+		},
```

Comment:
try this for a foreign key:

```
			if err := tx.AutoMigrate(&Consumer{}); err != nil {
				return err
			}
			return db.CreateFK(tx, db.FkMigration{
				"resource", "consumer", "consumer_id", "consumers(id)",
			})
```

Reply:
you'll need to export `db.FkMigration`

Reply:
actually, exporting makes a cyclic dependency.  instead, move the `CreateFK` func and `fkMigration` struct (https://github.com/openshift-online/maestro/blob/main/pkg/db/migrations.go#L42-L62) to this package's migration_structs.go.

Reply:
thank you @markturansky, done.